ledger-core/scripts/generate_insolar_configs.go

```diff
@@ -174,9 +178,45 @@ func prepareBootstrapVars() commonConfigVars {
 	return res
 }
 
+func generateBaseCloudConfig() {
+	if numVirtual < 1 {
+		check("cannot generate config", throw.New("virtual count must be more than zero"))
+	}
+
+	logConfig := configuration.Log{
+		Level:     debugLevel,
+		Adapter:   "zerolog",
+		Formatter: "json",
+	}
+
+	nodeConfigPathTmpl := ".artifacts/launchnet/discoverynodes/%d/insolard.yaml"
+	nodeConfigs := []string{}
+	for i := 1; i <= numVirtual; i++ {
+		nodeConfigs = append(nodeConfigs, fmt.Sprintf(nodeConfigPathTmpl, i))
+	}
+
+	conf := configuration.BaseCloudConfig{
+		Log:             logConfig,
+		NodeConfigPaths: nodeConfigs,
+	}
+
+	rawData, err := yaml.Marshal(conf)
+	check("Can't Marshal insolard config", err)
+
+	fileName := withBaseDir("configs/base_cloud.yaml")
+
+	err = createFileWithDir(fileName, string(rawData))
+	check("failed to create base cloud config: "+fileName, err)
+}
+
 func main() {
 	parseInputParams()
 
+	if baseCloud {
+		generateBaseCloudConfig()
```

Comment:
I'd move this to subcommand

Reply:
what for? It's also insolar node? Why one more file?

Reply:
because it calls different function and creates different config. So you don't need any other options, except for baseCloud. You can actually even skip numVirtual, and just glob search directory with configs.

Reply:
1. I don't see the problem, when one binary generates different configs with different options
2. I like the idea with explicit number of virtual more.
If it's principal - I can change. But I like current solution more

Reply:
1. I don't like the design of function that does 
```
func myFunc(condition bool) {
  if condition {
    return myOtherFunc()
  }
}

func myOtherFunc() {
}
```

It looks bad.

2. then you should not only check virtual, but check all kind of nodes params and read there configs, not only virtual

Reply:
1. I don't agree
2. Yes, here's ok. I will read all directory

Reply:
fixed
+ move to separate file